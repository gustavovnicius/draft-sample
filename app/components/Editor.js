import React, { Component } from 'react'
import Draft from 'draft-js'

const styles = {
  editor: {
    backgroundColor: '#fff',
    padding: '2em',
    boxShadow: '0px 0px 6px 1px rgba(0,0,0,0.18)',
    minWidth: '55vw',
    maxWidth: '55vw',
    minHeight: '50vh',
    cursor: 'text',
    margin: '0 0 2em 0'
  }
}

class Editor extends Component {
  render() {
    return (
      <div style={styles.editor} onClick={() => this.editor.focus()}>
        <Draft.Editor
          ref={editor => this.editor = editor}
          editorState={this.props.editorState}
          onChange={f => f}
        />
      </div>
    )
  }
}

export default Editor
