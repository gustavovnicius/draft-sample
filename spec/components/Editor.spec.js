import React from 'react';
import renderer from 'react-test-renderer';
jest.mock('draft-js')

import Editor from 'components/Editor';

describe('Editor', () => {
  it('renders', () => {
    const component = renderer.create(
      <Editor editorState={undefined} />
    )
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
