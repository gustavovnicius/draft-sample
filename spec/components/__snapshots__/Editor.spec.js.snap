exports[`Editor renders 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "boxShadow": "0px 0px 6px 1px rgba(0,0,0,0.18)",
      "cursor": "text",
      "margin": "0 0 2em 0",
      "maxWidth": "55vw",
      "minHeight": "50vh",
      "minWidth": "55vw",
      "padding": "2em",
    }
  } />
`;
